[cmake] fix ldflags transfer on addon depends build

AlwinEsch opened the pull request:
## Description

Before was only a "-DCMAKE_EXE_LINKER_FLAGS=-L${OUTPUT_DIR}/lib" given.
But by cross compile on some depends are library linker flags needed who give
infos about the OS (specially ffmpeg).

This change add the give of them in same way as already CFLAGS and CPPFLAGS
are given.

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch commented:
jenkins build this with addons please

Rechi commented on cmake/scripts/common/HandleDepends.cmake:
```diff
@@ -78,8 +78,10 @@ function(add_addon_depends addon searchpath)
           # make sure we create strings, not lists
           set(TMP_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_DEFINES}")
           set(TMP_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_DEFINES}")
+          set(TMP_EXE_LINKER_FLAGS "-L${OUTPUT_DIR}/lib ${CMAKE_EXE_LINKER_FLAGS} ${ARCH_DEFINES}")
```
Why are define flags needed in the linker flags?

AlwinEsch replied:
Make no sense, was only a nightly c&p from above and my Mr. Brainbrain was on other side of world.

jenkins build this with addons please

AlwinEsch committed:
[cmake] fix ldflags transfer on addon depends build

Before was only a "-DCMAKE_EXE_LINKER_FLAGS=-L${OUTPUT_DIR}/lib" given.
But by cross compile on some depends are library linker flags needed who give
infos about the OS (specially ffmpeg).

This change add the give of them in same way as already CFLAGS and CPPFLAGS
are given.

phunkyfish reviewed (approved):
LGTM.

AlwinEsch commented:
jenkins build this with addons please
